.travis.yml

```diff
@@ -0,0 +1,77 @@
+language: python
+
+# We need a newer version of zlib to build notmuch from source as is available
+# in precise.
+dist: trusty
+
+sudo: true
+
+python:
+  # We can add more version strings here when we support other python
+  # versions.
+  - "2.7"
+
+addons:
+  apt:
+    packages:
+      # The gpgme build files are needed by the gpgme python module
+      # (a dependency of alot).
+      - libgpgme11-dev
+      # The notmuch libs are needed to actually run alot.  The notmuch package
+      # should pull the correct libs but currently the available version is
+      # not compatible with alot, so we have to build from source.
+      #- notmuch
+      # Dependencies to build notmuch from source.
+      - libxapian-dev
+      - libtalloc-dev
+      - zlib1g-dev
+
+# Build notmuch and the python notmuch libs manually.  The versions of the
+# notmuch library and the python module which are available in the 12.04 and
+# 14.04 Ubuntu repos do not match and do not fullfill the version requirement
+# for alot.
+before_install:
+  - sudo apt-get update
+  # Until https://github.com/travis-ci/apt-package-whitelist/issues/3215 is
+  # resolved we have to install gmime with sudo.  Afterwards we can swtich to
+  # the container based infrastructure.
+  # gmime is needed to build notmuch from source.
+  - sudo apt-get -yqq install libgmime-2.6-dev
+  # Clone the notmuch repository and move into it.
+  - git clone git://notmuchmail.org/git/notmuch
```

Comment:
Observation: this is always going to test against notmuch's master branch. We ought to choose explicitly what to test against. One approach: test against both master and the release branch, possibly not blocking on failures from master (since it's a moving target anyway, but at least we hear about breaking before it actually lands).

I ended up doing something like this with my golang binding to notmuch:

https://github.com/zenhack/go.notmuch/blob/master/.travis.yml